src/simulation/LoadGenerator.cpp

```diff
@@ -463,52 +492,73 @@ LoadGenerator::handleFailedSubmission(TestAccountPtr sourceAccount,
 }
 
 std::vector<LoadGenerator::TestAccountPtr>
-LoadGenerator::checkAccountSynced(Application& app)
+LoadGenerator::checkAccountSynced(Application& app, bool isCreate)
 {
     std::vector<TestAccountPtr> result;
     for (auto const& acc : mAccounts)
     {
         TestAccountPtr account = acc.second;
-        auto currentSeqNum = account->getLastSequenceNumber();
-        auto reloadRes = loadAccount(account, app);
-        // reload the account
-        if (!reloadRes || currentSeqNum != account->getLastSequenceNumber())
+        auto accountFromDB = *account;
+
+        auto reloadRes = loadAccount(accountFromDB, app);
+        // For account creation, reload accounts from the DB
+        // For payments, ensure that the sequence number matches expected
+        // seqnum. Timeout after 20 ledgers.
+        if (!reloadRes && isCreate)
         {
-            CLOG(DEBUG, "LoadGen")
-                << "Account " << account->getAccountId()
-                << " is at sequence num " << currentSeqNum
-                << ", but the DB is at  " << account->getLastSequenceNumber();
+            CLOG(TRACE, "LoadGen") << "Account " << account->getAccountId()
+                                   << " is not created yet!";
             result.push_back(account);
         }
+        else if (reloadRes && !isCreate)
+        {
+            if (account->getLastSequenceNumber() !=
+                accountFromDB.getLastSequenceNumber())
+            {
+                CLOG(TRACE, "LoadGen") << "Account " << account->getAccountId()
+                                       << " is at sequence num "
+                                       << account->getLastSequenceNumber()
+                                       << ", but the DB is at  "
+                                       << accountFromDB.getLastSequenceNumber();
+                result.push_back(account);
+            }
+        }
     }
     return result;
 }
 
 void
-LoadGenerator::waitTillComplete()
+LoadGenerator::waitTillComplete(bool isCreate)
 {
     if (!mLoadTimer)
     {
         mLoadTimer = std::make_unique<VirtualTimer>(mApp.getClock());
     }
     vector<TestAccountPtr> inconsistencies;
-    inconsistencies = checkAccountSynced(mApp);
+    inconsistencies = checkAccountSynced(mApp, isCreate);
```

Comment:
Do we have a problem where `checkAccountSynced` is blocking for too long because it rescans all of `mAccounts` instead of the ones that didn't reach their goal? We may want to open an issue for this: if we run a simulation for millions of accounts, it may cause problems.

Reply:
I didn't encounter problems of this kind during benchmarking runs, but opened an issue as it's worth taking a look: https://github.com/stellar/stellar-core/issues/2223